tests/unit/phpunit/modules/InboundEmail/InboundEmailTest.php

```diff
@@ -22,51 +24,15 @@ function tempFileWithMode($mode)
     return $file;
 }
 
-class InboundEmailTest extends SuiteCRM\StateCheckerPHPUnitTestCaseAbstract
+class InboundEmailTest extends SuitePHPUnitFrameworkTestCase
 {
-
-    protected function storeStateAll()
-    {
-        // save state
-        $state = new SuiteCRM\StateSaver();
-        $state->pushTable('inbound_email_cache_ts');
-        $state->pushTable('inbound_email_autoreply');
-        $state->pushTable('inbound_email');
-        $state->pushTable('aod_index');
-        $state->pushTable('folders');
-        $state->pushTable('folders_subscriptions');
-        $state->pushTable('config');
-        $state->pushTable('email_cache');
-        $state->pushFile('config.php');
```

Comment:
This is probably the source of the problem, InboundEmailTest.php messes with the config.php.

Can we work around this somehow? Ideally, it wouldn't mess with the `config.php` at all because it would also screw it up when it fails.

We can fix this after merging this PR if we want to, I don't think it should be a blocker.

Reply:
I'm fairly certain it's the `saveInboundEmailSystemSettings` test specifically.
https://github.com/salesagility/SuiteCRM/blob/c9d2132df8db5955a0809301dacb9fb5db872b6c/modules/InboundEmail/InboundEmail.php#L5969-L5986

Reply:
Looking at the other uses of pushFile/popFile – besides the AOD Index which I'm not sure about – I don't think they actually need to be kept. TaskTest pops/pushes the config.php for no obvious reason, and CurrentLanguageTest does the same thing with the config_override.php, again for no apparent reason. So this is the only one that's really a problem.